library/std/src/io/buffered/bufreader/buffer.rs

```diff
@@ -93,11 +93,15 @@ impl Buffer {
         if self.pos >= self.filled {
             debug_assert!(self.pos == self.filled);
 
-            let mut readbuf = ReadBuf::uninit(&mut self.buf);
+            let mut buf: BorrowedBuf<'_> = (&mut *self.buf).into();
```

Comment:
`&mut *` is awkward either way, can't we use a method instead? So that this would be 
```rust
let mut buf = self.buf.borrow();
// or maybe something like `borrow_buf` so it doesn't conflict with `Borrow`? idk
```

Tbh I don't see how `Into` is useful here...

Reply:
`self.buf` is just a `Box<[MaybeUninit<u8>]>` here, so it would be a bit out of scope to for this PR to add `borrow` there.

> Tbh I don't see how Into is useful here...

We need some way to create a `BorrowBuf` from a `&mut [MaybeUninit<u8>]`, we could use a ctor function, but IMO `from`/`into` is more idiomatic.

Reply:
Oh, sorry, I thought `self.buf` was a specialized type. Nevermind then.